cocos/base/CCScheduler.cpp

```diff
@@ -457,10 +458,11 @@ void Scheduler::appendIn(_listEntry **list, const ccSchedulerFunc& callback, voi
     DL_APPEND(*list, listElement);
 
     // update hash entry for quicker access
-    tHashUpdateEntry *hashElement = (tHashUpdateEntry *)calloc(sizeof(*hashElement), 1);
+    tHashUpdateEntry *hashElement = new (std::nothrow) tHashUpdateEntry;
```

Comment:
??
tHashUpdateEntry is freed by `free`.
You could refer to https://github.com/newnon/cocos2d-x/blob/f903842c7c374b4a70bd5bd656c83d4d23efc58f/cocos/base/CCScheduler.cpp#L555

Reply:
Using `new` will cause memory problems.